test/evp_extra_test.c

```diff
@@ -2891,6 +2980,388 @@ static int test_names_do_all(void)
+    unsigned char outbuf[1024];
+    EVP_CIPHER_CTX *ctx = NULL;
+    EVP_CIPHER *type = NULL;
+    char *errmsg = NULL;
+
+    ctx = EVP_CIPHER_CTX_new();
+    if (ctx == NULL) {
+        errmsg = "CTX_ALLOC";
+        goto err;
+    }
+    type = EVP_CIPHER_fetch(testctx, "aes-128-cfb", testpropq);
+    if (type == NULL) {
+        errmsg = "CIPHER_FETCH";
+        goto err;
+    }
+    if (!EVP_CipherInit_ex(ctx, type, NULL, kCFBDefaultKey, iCFBIV, t->enc)) {
+        errmsg = "CIPHER_INIT";
+        goto err;
+    }
+    if (!EVP_CIPHER_CTX_set_padding(ctx, 0)) {
+        errmsg = "PADDING";
+        goto err;
+    }
+    if (!EVP_CipherUpdate(ctx, outbuf, &outlen1, t->input, t->inlen)) {
+        errmsg = "CIPHER_UPDATE";
+        goto err;
+    }
+    if (!EVP_CipherFinal_ex(ctx, outbuf + outlen1, &outlen2)) {
+        errmsg = "CIPHER_FINAL";
+        goto err;
+    }
+    if (!TEST_mem_eq(t->expected, t->expectedlen, outbuf, outlen1 + outlen2)) {
+        errmsg = "WRONG_RESULT";
+        goto err;
+    }
+    if (!EVP_CipherInit_ex(ctx, NULL, NULL, NULL, NULL, -1)) {
+        errmsg = "CIPHER_REINIT";
+        goto err;
+    }
+    if (!EVP_CipherUpdate(ctx, outbuf, &outlen1, t->input, t->inlen)) {
+        errmsg = "CIPHER_UPDATE (reinit)";
+        goto err;
+    }
+    if (!EVP_CipherFinal_ex(ctx, outbuf + outlen1, &outlen2)) {
+        errmsg = "CIPHER_FINAL (reinit)";
+        goto err;
+    }
+    if (!TEST_mem_eq(t->expected, t->expectedlen, outbuf, outlen1 + outlen2)) {
+        errmsg = "WRONG_RESULT (reinit)";
+        goto err;
+    }
+    testresult = 1;
+ err:
+    if (errmsg != NULL)
+        TEST_info("test_evp_reset %d: %s", idx, errmsg);
+    EVP_CIPHER_CTX_free(ctx);
+    EVP_CIPHER_free(type);
+    return testresult;    
+}
+
+typedef struct {
+    const unsigned char *iv1;
+    const unsigned char *iv2;
+    const unsigned char *expected1;
+    const unsigned char *expected2;
+    const unsigned char *tag1;
+    const unsigned char *tag2;
+    size_t ivlen1;
+    size_t ivlen2;
+    size_t expectedlen1;
+    size_t expectedlen2;
+} TEST_GCM_IV_REINIT_st;
+
+static const TEST_GCM_IV_REINIT_st gcm_reinit_tests[] = {
+    {
+        iGCMResetIV1, iGCMResetIV2, gcmResetCiphertext1, gcmResetCiphertext2,
+        gcmResetTag1, gcmResetTag2, sizeof(iGCMResetIV1), sizeof(iGCMResetIV2),
+        sizeof(gcmResetCiphertext1), sizeof(gcmResetCiphertext2)
+    },
+    {
+        iGCMResetIV2, iGCMResetIV1, gcmResetCiphertext2, gcmResetCiphertext1,
+        gcmResetTag2, gcmResetTag1, sizeof(iGCMResetIV2), sizeof(iGCMResetIV1),
+        sizeof(gcmResetCiphertext2), sizeof(gcmResetCiphertext1)
+    }
+};
+
+static int test_gcm_reinit(int idx)
+{
+    int outlen1, outlen2, outlen3;
+    int testresult = 0;
+    unsigned char outbuf[1024];
+    unsigned char tag[16];
+    const TEST_GCM_IV_REINIT_st *t = &gcm_reinit_tests[idx];
+    EVP_CIPHER_CTX *ctx = NULL;
+    EVP_CIPHER *type = NULL;
+    char *errmsg = NULL;
+
```

Comment:
maybe do
````
size_t taglen = sizeof(tag);
````
And replace the majic number 16 in the code below

Reply:
Done.